src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+		prev.selectedNumber === next.selectedNumber
+	);
+}
+
+const TableBody = props => {
+	return <tbody className={props.className}>{props.rows}</tbody>;
+};
+
+export const MemoTableBody = React.memo(TableBody, propsAreEqualBody);
+
+const TableCell = withDeferredTooltip(
+	React.forwardRef((props, ref) => {
+		return (
+			<td className={props.className} ref={ref} {...props}>
+				{props.value}
+			</td>
+		);
+	}),
+);
+
+// We will assume for now that cell will never change value so we want to avoid rerendering as much as possible
+export const MemoTableCell = React.memo(TableCell, () => true);
+
+const buildRowCheckbox = (classes, key, selectionHandlers) => {
+	const rowSelected = selectionHandlers.isSelected(key);
+
+	return (
+		<td className={classes.tableCellSelect} selected={rowSelected}>
+			<CheckboxMui
+				className={classes.rowSelectCheckbox}
+				checked={rowSelected}
+				onChange={event => selectionHandlers.selectionHandler(event, key)}
+			/>
+		</td>
+	);
+};
+
+const buildTableCheckbox = (classes, tableSelectionStatus, selectionMethods) => {
+	return (
+		<th key="tableSelector" className={classes.headerCellSelect}>
+			<CheckboxMui
+				className={classes.rowSelectCheckbox}
+				checked={tableSelectionStatus === tableSelectionMode.all}
+				indeterminate={tableSelectionStatus === tableSelectionMode.indeterminate}
+				onChange={event => selectionMethods.selectionHandler(event, null)}
+			/>
+		</th>
+	);
+};
+
+const StickerTableHeader = React.forwardRef((props, refHeader) => (
+	<div key="stickyHeader" className={props.classes.stickyHeaderHead}>
+		<TableMui className={props.classes.stickyHeaderTable}>
+			<TableHead ref={refHeader}>
+				<TableRow>{props.tableHeaders}</TableRow>
+			</TableHead>
+		</TableMui>
+		<div className={props.classes.stickyHeaderTableScroll} />
+	</div>
+));
+
+const buildTableHeaders = (
+	headers,
+	classes,
+	selectMode,
+	tableSelectionStatus,
+	selectionMethods,
+) => {
+	const tableCheckbox =
+		selectMode === true
+			? [buildTableCheckbox(classes, tableSelectionStatus, selectionMethods)]
+			: [];
+
+	return tableCheckbox.concat(
+		headers.map((header, index) => (
+			<th key={index} className={classes.headerCell} style={header.cellStyle}>
+				{header.cellElement}
+			</th>
+		)),
+	);
+};
+
+const buildTableRows = (rows, classes, selectMode, onRowClick, selectionHandlers) => {
+	const onClick = (evt, row) => {
+		if (evt.target.tagName !== "INPUT") {
+			onRowClick(evt, row.element);
+		}
+	};
+
+	return rows.map(row => (
+		<MemoTableRow
+			className={classes.tableRow}
+			key={row.key}
+			onClick={evt => onClick(evt, row)}
+			selected={selectionHandlers.isSelected(row.key)}
+		>
+			{selectMode === true ? buildRowCheckbox(classes, row.key, selectionHandlers) : null}
+			{row.columns.map((cell, cellIndex) => (
+				<MemoTableCell
+					className={classes.tableCell}
+					key={cellIndex}
+					value={cell.cellElement}
+					titleValue={cell.title}
+					style={cell.cellStyle}
```

Comment:
What style property will do here? it's not react right?
You can use several classes inside className. To mix classes you can use classnames package (you can find sample inside ExpansionPanel.js)

Reply:
Ditto